tests/sai_qualify/conftest.py

```diff
@@ -74,23 +82,39 @@ def start_sai_test_container(duthost, creds, deploy_sai_test_container, request)
 
 
 @pytest.fixture(scope="module")
-def deploy_sai_test_container(duthost, creds, stop_other_services, prepare_saiserver_script, request):
+def deploy_sai_test_container(duthost, creds, stop_other_services, prepare_saiserver_script, request, localhost):
     """
         Deploys a sai test container.
     """
+    sai_thrift_version = ""
+    os_version = duthost.os_version
```

Comment:
this os_version already exists in the duthost and it only used in the private method. we can use it from duthost, and transform this os_version in the place where actual needed
for those logic to transform the data, please encap them in a method.